codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/customize/CoreCodegenDecorator.kt

```diff
@@ -94,6 +95,11 @@ interface CoreCodegenDecorator<CodegenContext> {
      * Extra sections allow one decorator to influence another. This is intended to be used by querying the `rootDecorator`
      */
     fun extraSections(codegenContext: CodegenContext): List<AdHocCustomization> = listOf()
+
+    /**
+     * Hook for customizing symbols by inserting an additional symbol provider.
+     */
+    fun symbolProvider(base: RustSymbolProvider, model: Model): RustSymbolProvider = base
```

Comment:
I don't think you need `model` here—model is already available on the inner provider